cache_images/ubuntu_packaging.sh

```diff
@@ -27,21 +27,53 @@ if [[ "$OS_RELEASE_VER" -le 2004 ]]; then
 fi
 
 # The OpenSuse Open Build System must be utilized to obtain newer
-# development versions of podman/buildah/skopeo & dependencies,
-# in order to support upstream (i.e. bleeding-edge) development and
-# automated testing.  These packages are not otherwise intended for
-# end-user consumption.
-VERSION_ID=$(source /etc/os-release; echo $VERSION_ID)
-# Overview: https://build.opensuse.org/project/show/devel:kubic:libcontainers:unstable
-REPO_URL="https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_$VERSION_ID/"
-GPG_URL="https://download.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_$VERSION_ID/Release.key"
-
-echo "deb $REPO_URL /" | ooe.sh $SUDO \
-    tee /etc/apt/sources.list.d/devel:kubic:libcontainers:unstable:ci.list
-curl --fail --silent --location --url "$GPG_URL" | \
-    gpg --dearmor | \
-    $SUDO tee /etc/apt/trusted.gpg.d/devel_kubic_libcontainers_unstable_ci.gpg &> /dev/null
-
+# development versions of some tools. This helper sets up config
+# files for apt to fetch packages from OBS. We can be called with
+# a variable number of arguments.
+function setup_obs() {
+    # Version of ubuntu, e.g., 22.04
+    local xubuntu_version
+    xubuntu_version="xUbuntu_$(source /etc/os-release; echo $VERSION_ID)"
+
+    local base_url="https://download.opensuse.org/repositories/devel"
+
+    # Assemble the .deb repo URL by appending colon-slash-item for each arg
+    local repo_url="$base_url"
+    local repo_file="/etc/apt/sources.list.d/devel"
+    for i in "$@"; do
+        repo_url+=":/$i"
+        repo_file+=":$i"
+    done
+    repo_url+="/${xubuntu_version}/"
+    repo_file+=":ci.list"
+    echo "deb $repo_url /" | ooe.sh $SUDO tee "$repo_file"
+
+    # GPG key URL is similar to .deb repo, but just colons, no slashes
+    local gpg_url="$base_url"
+    local gpg_file="/etc/apt/trusted.gpg.d/devel"
+    for i in "$@"; do
+        gpg_url+=":$i"
+        gpg_file+="_$i"
+    done
+    gpg_url+="/${xubuntu_version}/Release.key"
+    gpg_file+="_ci.gpg"
+    curl --fail --silent --location --url "$gpg_url" | \
+        gpg --dearmor | \
+        $SUDO tee "$gpg_file" &> /dev/null
+}
```

Comment:
For consistency, would it be possible to move this into `lib.sh`?

Reply:
No, because it's only useful for ubuntu, and lib.sh is used for non-ubuntu.